routes/weatherProviders/DarkSky.ts

```diff
@@ -49,16 +83,16 @@ async function getDarkSkyWeatherData( coordinates: GeoCoordinates ): Promise< We
 
 		region: "",
 		city: "",
-		minTemp: Math.floor( forecast.daily.data[ 0 ].temperatureLow ),
-		maxTemp: Math.floor( forecast.daily.data[ 0 ].temperatureHigh ),
+		minTemp: Math.floor( forecast.daily.data[ 0 ].temperatureMin ),
+		maxTemp: Math.floor( forecast.daily.data[ 0 ].temperatureMax ),
 		precip: forecast.daily.data[ 0 ].precipIntensity * 24,
-		forecast: []
+		forecast: [ ]
 	};
 
 	for ( let index = 0; index < forecast.daily.data.length; index++ ) {
 		weather.forecast.push( {
-			temp_min: Math.floor( forecast.daily.data[ index ].temperatureLow ),
-			temp_max: Math.floor( forecast.daily.data[ index ].temperatureHigh ),
+			temp_min: Math.floor( forecast.daily.data[ index ].temperatureMin ),
+			temp_max: Math.floor( forecast.daily.data[ index ].temperatureMax ),
 			date: forecast.daily.data[ index ].time,
 			// TODO set this
 			icon: "",
```

Comment:
I think we need to translate weather provider specific `icons` into a normalised set of `conditions` that we then report onward to App. We could use MDI Icons for App. Below is a table from Home Assistant showing OWM/DarkSky/MDI mapping:

| MDI Icons | DarkSky Icons | OWM Icons | OWM Condition |
| --- | --- | ---- | ---- |
| 'weather-sunny' | 'clear-day' | 01d | [800] |
| 'weather-night' | 'clear-night' | 01n |  |
| 'weather-rainy' | 'rain' | 09d, 09n | [300, 301, 302, 310, 311, 312, 313, 500, 501, 520, 521] |
| 'weather-pouring' | 10d, 10n | [504, 314, 502, 503, 522] |
| 'weather-snowy' | 'snow' | 13d, 13n | [600, 601, 602, 611, 612, 620, 621, 622] |
| 'weather-snowy-rainy' | 'sleet' | [511, 615, 616] |
| 'weather-windy' | 'wind' | [905, 951, 952, 953, 954, 955, 956, 957] |
| 'weather-windy-variant' | [958, 959, 960, 961] |
| 'weather-fog' | 'fog' | 50d, 50n | [701, 741] |
| 'weather-cloudy' | 'cloudy' | 04d, 04n | [803, 804] |
| 'weather-partlycloudy' | 'partly-cloudy-day' | 02d, 02n, 03d, 03n | [801, 802] |
| 'weather-partlycloudy' | 'partly-cloudy-night' |  |
| 'weather-hail' | 'hail' | [906] |
| 'weather-lightning' | 'thunderstorm' |  |
| 'weather-lightning-rainy' | 11d, 11n | [210, 211, 212, 221] |
| 'weather-hurricane' | 'tornado' |  |

Unmapped OWM
711 - Smoke, 721 - Haze, 731 - Dust, 751 - Sand, 761 - Dust, 762 - Ash, 771 - Squall

Reply:
This is fantastic and exactly what I want to achieve as we add more weather sources. This is a great start and I will try to look into this ASAP. Keep in mind as we start making these breaking changes older clients will no longer show an icon, etc. This isn't an issue for the majority of users as updates occur fairly easily/automatically on most platforms.

Reply:
Moved to Issue #46